File: sdk/auto-openai.ts
// Auto-instrumentation (TypeScript) — one import
import '@hippocortex/sdk/auto'
import OpenAI from 'openai'

const openai = new OpenAI()

const response = await openai.chat.completions.create({
  model: 'gpt-4o',
  messages: [{ role: 'user', content: 'Deploy payments to staging' }],
})
console.log(response.choices[0].message.content)

File: sdk/auto-openai.py
"""Auto-instrumentation — one import adds memory to all OpenAI calls."""
import hippocortex.auto  # This is the only change needed
from openai import OpenAI

client = OpenAI()

response = client.chat.completions.create(
    model="gpt-4o",
    messages=[{"role": "user", "content": "Deploy payments to staging"}],
)
print(response.choices[0].message.content)
